utils/auth.js

```diff
@@ -0,0 +1,93 @@
+import { Component } from "react";
+import Router from "next/router";
+import nextCookie from "next-cookies";
+import cookie from "js-cookie";
+
+export const login = async ({ token }) => {
+  cookie.set("token", token, { expires: 1 });
+  // localStorage.setItem('token', token)
+  Router.push("/logged");
+};
+
+export const signup = async ({ token }) => {
+  cookie.set("token", token, { expires: 1 });
+  // localStorage.setItem('token', token)
+  Router.push("/");
+};
+
+export const logout = () => {
+  cookie.remove("token");
+  // to support logging out from all windows
+  window.localStorage.setItem("logout", Date.now());
+  // window.localStorage.removeItem('token')
+  Router.push("/login");
+};
+
+export const isAuthenticated = !!cookie.get("token");
```

Comment:
should this be a function ?

Reply:
maybe a hook ?

Reply:
J'ai mis en place un useAuth pour gérer l'ensemble de l'authentification. Il me reste encore quelques tests à faire je push ça rapidement.